clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -559,6 +559,8 @@ void EmitAssemblyHelper::CreatePasses(legacy::PassManager &MPM,
   PMBuilder.PrepareForThinLTO = CodeGenOpts.PrepareForThinLTO;
   PMBuilder.PrepareForLTO = CodeGenOpts.PrepareForLTO;
   PMBuilder.RerollLoops = CodeGenOpts.RerollLoops;
+  if (TargetTriple.isOSBinFormatRepo())
```

Comment:
> 
> 
> The subject of the change is “Modify the RepoPruning pass…” so I’m a little surprised to see the pass manager being modified. What’s the reason? 
>
  When I added the repo related passes in llvm backend (three year ago), I was not familiar with the llvm pass manager. I made the pass manager invoke the repo related passes on all platform but only run these passes when targeted on the Repo. Now, I find a way to invoke the repo related passes only targeted on the repo.  
>
> It seems like this might be something to address separately, perhaps?
>
Good point, the initial commit is separated into three different commits. The commit (0b2a89527db836770fa571a4e89) addresses this issue.